common/main/ai.h

```diff
@@ -182,6 +182,7 @@ static inline void force_dump_ai_objects_all(const char *msg)
 namespace dsx {
 void start_boss_death_sequence(vobjptr_t objp);
 extern void ai_init_boss_for_ship(void);
+extern void boss_init_all_segments(objptridx_t boss_objnum);
```

Comment:
Both callers of `boss_init_all_segments` pass a `vobjptridx_t`.  While it can be transparently converted to `objptridx_t`, retaining its original type would document that it must be a valid object.  That would then remove the need for one branch of the new `Assert` inside `init_boss_segments`.